simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -1376,6 +1377,11 @@ auto HdMapUtils::getVectorFromPose(const geometry_msgs::msg::Pose & pose, const
   return vector;
 }
 
+auto HdMapUtils::isInIntersection(const lanelet::Id lanelet_id) const -> bool
+{
+  return lanelet_map_ptr_->laneletLayer.get(lanelet_id).hasAttribute("turn_direction");
```

Comment:
In my feeling, using `“turn_direction”` to simply get information about whether a lanelet is at an intersection is most proper. 

However, I would (at least) add a comment to this method about the fact that `“turn_direction”` **must be set** in the lanelet that is at the intersection, but **can be set** on other lanelets. As can be read [here](https://github.com/autowarefoundation/autoware_common/blob/main/tmp/lanelet2_extension/docs/lanelet2_format_extension.md#turn-directions):

> Users must add "turn_direction" tags to lanelets within intersections to indicate vehicle's turning direction. You do not need this tags for lanelets that are not in intersections.

Of course, I realize that in practice it is almost never set, but it can happen that a lanelet that is not at an intersection has `turn_direction==“straight”` set - in that case this method will return true for that lanelet, which is incorrect and can be confusing. Let me know whether you think the same. :bow: 

Also, I would consider how to handle crosswalks here, since they never have a `“turn_direction”`. Maybe it would make sense to return `“true”` for each pedestrian crosswalk, since it by its nature always intersects with the road? :bow:

Reply:
Sorry for sending [Japanese documentation](https://docs.pilot.auto/reference-design/common/map-requirements/vector-map-requirements/category_intersection/#vm-03-02-%E3%83%AC%E3%83%BC%E3%83%B3%E3%81%AE-turn_direction-%E3%81%A8-virtual), but lanelet2 which used by Autoware in intersection always has "turn_direction" attribute.